CHANGELOG.md

```diff
@@ -3,6 +3,7 @@
 ### Bug fixes
 
 * Fixed a crash when calling Table.toString() in debugger (#2429).
+* Fixed a bug when migration loses tracking of primary key field in `RealmObjectSchema.removeField()` and `RealmObjectSchema.renameField()` (#2829).
```

Comment:
Fixed a bug that could cause Realm to loose track of the correct primary key when using `RealmObjectSchema.removeField()` and `RealmObjectSchema.renameField()`